src/main/java/net/obvj/agents/conf/ConfigurationHolder.java

```diff
@@ -86,6 +86,12 @@ private static TypeSafeConfigurationContainer<Configuration> defaultContainer()
     private void fillAuxiliaryObjects()
     {
         Configuration config = container.getBean();
+        // Set empty as defaults if none of .xml,.json or .yaml are supplied and caller will invoke scan packages programatically.
+        if (config == null) {
+            agentsByClassName = new HashMap<>();
```

Comment:
The compilation is failing due to a missing import for `HashMap`.
Please try to use `Collections.emptyMap()` instead.


````
Error:  Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.1:compile (default-compile) on project agents: Compilation failure
Error:  /home/runner/work/agents/agents/src/main/java/net/obvj/agents/conf/ConfigurationHolder.java:[91,37] cannot find symbol
Error:    symbol:   class HashMap
Error:    location: class net.obvj.agents.conf.ConfigurationHolder
Error:  -> [Help 1]
````

Reply:
My bad, i edited in Web IDE. Took a pull, compiled and squashed commits.
Please check :)